word_break.java

```diff
@@ -0,0 +1,56 @@
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+class GFG
+ {
+    public void wordBreak(String sentence,String[] dictionary,int n)
+    {
+        wordBreakUtil(sentence,dictionary,n,"");
+    }
+    
+    public void wordBreakUtil(String sentence,String[] dictionary,int n,String result)
+    {
+        for(int j=1;j<=n;j++)
+        {
+            String prefix=sentence.substring(0,j);
+            if(isDictionarryContain(prefix,dictionary))
+            {
+                
+                if(j==n)
+                {
+                    result=result+prefix;
+                    System.out.print("("+result+")");
+                    return;
+                }
+                wordBreakUtil(sentence.substring(j,n),dictionary,n-j,result+prefix+" ");
+            }
+        }
+    }
+    private boolean isDictionarryContain(String prefix,String []dictionary)
+    {
+        for(int i=0;i<dictionary.length;i++)
+        {
+            if(prefix.equals(dictionary[i]))
+                return true;
+        }
+        return false;
+    }
+	public static void main (String[] args)
+	 {
+	    Scanner sc=new Scanner(System.in);
+	    int test=sc.nextInt();
+	    while(test>0)
```

Comment:
Couple of notes here:
1) this seems like you should be using a for loop instead of a while loop since you are running it a known number of times
2)  A similar issue with my comment about "n" above.  Especially since you use this variable name for two different things (string and dictionary length) this should have two different names
3) Is it necessary to have a new GFG every time? It seems like you could have created one outside of the loop.  Also, since GFG does not have any properties or fields, should the methods be static?
4) The indentation is off here